### DISCUSSION TITLE
Are bar and spiral arm potentials added by default in MW Potential 2014?

### DISCUSSION THREAD

**Discussion by CakmakHi** [jobovy/galpy#513] [General]
I am studying on open star clusters and using Galpy's MWPotential2014 for orbital calculations. Although I am using the potential with defaults values, I don't know which potentials are included. Are bar and spiral arm potentials set to default in MWPotential 2014? Or will we add these potentials with parameters that we set? If we are going to do it ourselves, how should we add these potentials with parameters? Thanks from now for any response.

**Comment by jobovy** [comment]
``MWPotential2014`` does not include any bar or spiral structure, it's a simple axisymmetric model consisting of a spherical bulge, Miyamoto-Nagai disk, and an NFW dark-matter halo, It's described in detail in [the galpy paper](https://ui.adsabs.harvard.edu/abs/2015ApJS..216...29B/abstract). To add a bar or spiral structure, you can add extra potential components representing those from those implemented in ``galpy``:

https://docs.galpy.org/en/latest/reference/potential.html#spiral-bar-other-triaxial-and-miscellaneous-potentials

For example, you could add a bar as a ``DehnenBarPotential`` and spiral arms as a ``SpiralArmsPotential``. Both of these modulate the axisymmetric density, so then can be added to ``MWPotential2014`` (or the other MW models) without changing their mass. But you'd have to decide on reasonable parameters for them yourself, ``galpy`` currently does not have a default set of MW bar and spiral parameters.

**Comment by stevenalfonso** [comment]
Hello, 
I am trying to add a bar to the MWPotential2014 as pot = MWPotential2014+SoftenedNeedleBarPotential(), and then convert it to AMUSE mwp_amuse=to_amuse(pot, reverse=True). However, when added to the AMUSE integration, there is an error due to potential shape, I guess. 
Particularly: ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (4,) + inhomogeneous part.

This error comes when evolving the gravity code, as following the example provided in the Conversion to AMUSE potentials section in Galpy documentation. 
How can I solve this problem?
Thanks!

**Reply by jobovy** [reply]
I think this should work, can you give a fuller error message and if possible, a simple example program that gives this problem? This would also be better as a separate Discussion.
